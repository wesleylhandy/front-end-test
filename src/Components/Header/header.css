.header {
	background-color: #636463;
	padding: 1rem 1rem 0 1rem;
	color: white;
	display: flex;
}

.header__logo {
	height: 5rem;
	display: block;
}

.header__title {
	font-size: 1.5em;
	flex: 1;
}

.header__title--unread-count {
	background: #ee3524;
	color: white;
	border-radius: 2px;
	border: solid 1px white;
	font-size: 1rem;
	line-height: 1rem;
	height: 1rem;
	padding: 0.2rem 0.3rem;
	margin-left: 0.3rem;
}
