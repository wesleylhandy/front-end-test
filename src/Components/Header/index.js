import React from "react";
import PropTypes from "prop-types";
import "./header.css";
import logo from "../../SVG/logo.svg";

const Header = ({ unreadCount = 0 }) => (
	<header className="header">
		<img src={logo} className="header__logo" alt="logo" />
		<h1 className="header__title">Notes Viewer Test App</h1>
		<div className="header__unread-count">
			Unread:
			<span className="header__unread-count--count">{unreadCount}</span>
		</div>
	</header>
);

Header.propTypes = {
	unreadCount: PropTypes.number.isRequired,
};

export default Header;
