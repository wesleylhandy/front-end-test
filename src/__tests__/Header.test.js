import React from "react";
import Header from "../Components/Header";
import { create } from "react-test-renderer";

it("renders correctly", () => {
	const tree = create(<Header unreadCount={0} />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("updates unread count when props change", () => {
	const tree = create(<Header unreadCount={20} />).toJSON();
	expect(tree).toMatchSnapshot();
});
