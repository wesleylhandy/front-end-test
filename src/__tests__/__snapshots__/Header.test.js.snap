// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="header"
>
  <img
    alt="logo"
    className="header__logo"
    src="logo.svg"
  />
  <h1
    className="header__title"
  >
    Notes Viewer Test App
  </h1>
  <div
    className="header__unread-count"
  >
    Unread:
    <span
      className="header__unread-count--count"
    >
      0
    </span>
  </div>
</header>
`;

exports[`updates unread count when props change 1`] = `
<header
  className="header"
>
  <img
    alt="logo"
    className="header__logo"
    src="logo.svg"
  />
  <h1
    className="header__title"
  >
    Notes Viewer Test App
  </h1>
  <div
    className="header__unread-count"
  >
    Unread:
    <span
      className="header__unread-count--count"
    >
      20
    </span>
  </div>
</header>
`;
